lib/Graph/NetworkComparator.cpp

```diff
@@ -0,0 +1,365 @@
+      inputToFeed = singleLayerMod.createConstant(constNode->getName(), payLoad,
+                                                  constNode->getLayout());
+      singleLayerInputMap[originalNode.getInputName(idx)] = &payLoad;
+    } else {
+      if (!llvm::isa<Placeholder>(inputNode)) {
+        // If the input is placeholder the name stays the
+        // same since these don't get hooked.
+        std::string outputName = inputNode->getOutputName(resNo);
+        hookedPlaceholderName =
+            outputName + "_" + hookedPlaceholderName + "_hook";
+      }
+      DEBUG_GLOW(LOG(INFO) << "\t\tInput name: " << hookedPlaceholderName
+                           << " NodeType: " << inputNode->getKindName()
+                           << "\n");
+      Placeholder *PH =
+          refHookedBindigs_.getPlaceholderByName(hookedPlaceholderName);
+      DCHECK(PH) << "Placeholder not found in the hooked bindings";
+      Tensor *payloadTensor = refHookedBindigs_.get(PH);
+      Placeholder *singleLayerPH = singleLayerMod.createPlaceholder(
+          PH->getType(), PH->getName(), PH->isTraining(), PH->getLayout());
+      singleLayerBindings.allocate(singleLayerPH);
+      singleLayerBindings.get(singleLayerPH)->assign(payloadTensor);
+      inputToFeed = singleLayerPH;
+      singleLayerInputMap[originalNode.getInputName(idx)] = payloadTensor;
+    }
+    singleLayerNode->setNthInput(idx, inputToFeed);
+  }
+}
+
+void IntermediateLayerComparator::runAndGetoutputSingleLayer(
+    ExecutionEngine &singleLayerExecEng,
+    PlaceholderBindings &singleLayerBindings, Node *singleLayerNode,
+    std::unordered_map<std::string, Tensor *> &singleLayerOutputs,
+    std::unordered_map<std::string, Tensor *> &refOutputs) {
+  std::list<SaveNode *> singleLayerSaveNodes;
+  std::list<Placeholder *> singleLayerOutputPHs;
+  hookSingleNodeInPlace(*singleLayerNode, singleLayerSaveNodes,
+                        singleLayerOutputPHs);
+  singleLayerBindings.allocate(singleLayerOutputPHs);
+  DEBUG_GLOW(LOG(INFO) << "\t\tSingle layer network"
+                       << singleLayerNode->getParent()->dumpDAG());
+  singleLayerExecEng.compile(CompilationMode::Infer);
+  singleLayerExecEng.run(singleLayerBindings);
+  for (Placeholder *PH : singleLayerOutputPHs) {
+    singleLayerOutputs[PH->getName()] = singleLayerBindings.get(PH);
+    Placeholder *refPH = refHookedBindigs_.getPlaceholderByName(PH->getName());
+    refOutputs[PH->getName()] = refHookedBindigs_.get(refPH);
+  }
+}
+
+bool IntermediateLayerComparator::testSingleLayer(const Node *node) {
+  bool pass = true;
+  std::unordered_map<std::string, Tensor *> singleLayerInputMap;
+  ExecutionEngine singleLayerExecEng(EETestNet_.getBackendName());
+  PlaceholderBindings singleLayerBindings;
+  Module &singleLayerMod = singleLayerExecEng.getModule();
+  Function *singleLayerFunc = singleLayerMod.createFunction(node->getName());
+  Node *singleLayerNode = node->clone();
+  std::string layerName = node->getName();
```

Comment:
`llvm::StringRef`?

Reply:
StringRef only  has overrides with the llvm::outs output streams :(